analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Constants.cs

```diff
@@ -0,0 +1,84 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn
+{
+    public partial class RoslynSymbolicExecutionTest
+    {
+        [TestMethod]
+        [DataRow("true")]
+        [DataRow("false")]
+        public void PreProcess_TrueLiteral_BooleanConstraint(string value)
+        {
+            var expectedConstraint = GetBooleanConstraint(value);
+
+            var validator = SETestContext.CreateCS($"var a = {value};", new EmptyTestCheck()).Validator;
+            validator.ValidateOrder(
+                $"LocalReference: a = {value} (Implicit)",
+                $"Literal: {value}",
+                $"SimpleAssignment: a = {value} (Implicit)");
+            validator.Validate($"Literal: {value}", x => x.State[x.Operation].HasConstraint(expectedConstraint).Should().BeTrue());
+            validator.Validate($"SimpleAssignment: a = {value} (Implicit)", x => x.State[x.Operation].HasConstraint(expectedConstraint).Should().BeTrue());
+        }
+
+        [TestMethod]
+        [DataRow("1", "true")]
+        [DataRow("2", "false")]
+        public void PreProcess_EqualityCheck_BooleanConstraint(string constant, string value)
+        {
+            var expectedConstraint = GetBooleanConstraint(value);
+
+            var validator = SETestContext.CreateCS($"bool a = 1 == {constant};", new EmptyTestCheck()).Validator;
+            validator.ValidateOrder(
+                $"LocalReference: a = 1 == {constant} (Implicit)",
+                "Literal: 1",
+                $"Literal: {constant}",
+                $"BinaryOperator: 1 == {constant}",
+                $"SimpleAssignment: a = 1 == {constant} (Implicit)");
+            validator.Validate($"BinaryOperator: 1 == {constant}", x => x.State[x.Operation].HasConstraint(expectedConstraint).Should().BeTrue());
+            validator.Validate($"SimpleAssignment: a = 1 == {constant} (Implicit)", x => x.State[x.Operation].HasConstraint(expectedConstraint).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void PreProcess_EqualityCheck_ConstantPropagation()
+        {
+            var validator = SETestContext.CreateCS("const int zero = 0; bool a = 1 == zero;", new EmptyTestCheck()).Validator;
```

Comment:
For completeness, where `f` and `t` indicates expectation.
```suggestion
            var validator = SETestContext.CreateCS("const int zero = 0; bool f = 1 == zero, t = 0 == zero;", new EmptyTestCheck()).Validator;
```